conftest.py

```diff
@@ -16,7 +16,7 @@ def _crawler(extended_settings={}):
         }
         settings.update(extended_settings)
         crawler = Crawler(Spider, settings=settings)
-        crawler.spider = Spider("dummy")
+        crawler.spider = Spider.from_crawler(crawler, name="dummy")
```

Comment:
If you want to make things more future-proof, you can use https://github.com/scrapy/scrapy/blob/8fbebfa943c3352f5ba49f46531a6ccdd0b52b60/scrapy/utils/misc.py#L143